lib/components/hike_screen_widget.dart

```diff
@@ -218,25 +212,37 @@ class HikeScreenWidget {
                   itemCount: model.hikers.length,
                   itemBuilder: (BuildContext context, int index) {
                     return ListTile(
-                      onTap: () {
-                        model.hikers[index].id == userConfig.currentUser.id
-                            ? Fluttertoast.showToast(msg: 'Yeah, that\'s you')
-                            : model.beacon.leader.id ==
-                                    userConfig.currentUser.id
-                                ? model.relayBeacon(model.hikers[index])
-                                : Fluttertoast.showToast(
-                                    msg: 'You dont have beacon to relay');
+                      onLongPress: () async {
+                        print(model.hikers[index].id);
+                        if (model.hikers[index].id == userConfig.currentUser.id)
+                          Fluttertoast.showToast(msg: 'Yeah, that\'s you');
+                        else {
+                          if (model.beacon.leader.id ==
+                              userConfig.currentUser.id) {
+                            await databaseFunctions.init();
+                            final changedLeader =
+                                databaseFunctions.changeLeader(
+                                    model.beacon.id, model.hikers[index].id);
+                            if (changedLeader != null)
+                              model.relayBeacon(model.hikers[index].toString(),
+                                  model.hikers[index].id);
+                          } else {
+                            Fluttertoast.showToast(
+                                msg: 'You dont have beacon to relay');
+                          }
+                        }
```

Comment:
It would be better to keep entire logic thing in model only. 
Also this issue follows  the issue #136, i.e after relaying the leader, location is not updated anywhere in the hike screen for follower.

Reply:
So should I wait for issue #136 to get resolved first or create another future issue to look upon the changes in #136 and then implement it?